src/main/java/org/jenkinsci/remoting/protocol/impl/NIONetworkLayer.java

```diff
@@ -147,12 +147,12 @@ public void ready(boolean accept, boolean connect, boolean read, boolean write)
                             while (recv.hasRemaining()) {
                                 onRead(recv);
                             }
-                            if (in.isOpen() && recvKey.isValid()) {
-                                getIoHub().addInterestRead(recvKey);
-                            } else {
-                                recvKey.cancel();
-                                onRecvClosed();
-                            }
+                        }
+                        if (in.isOpen() && recvKey.isValid()) {
```

Comment:
Hmmm... Why does the current [`count == -1`](https://github.com/jenkinsci/remoting/pull/115/files#diff-5bd66617da2d4d7a7625a042f0297afaR136) check not cancel the `recvKey`?